providers/default.rb

```diff
@@ -1,16 +1,28 @@
 use_inline_resources
 
 action :update do
+  dir        = new_resource.base_name.split('/')[0..-2].join('/')
+  version    = new_resource.version
+  push_event = new_resource.push_to_datadog
+  event_name = new_resource.datadog_event_name
+  event_text = new_resource.datadog_event_text
+  event_key  = new_resource.datadog_event_key
+
```

Comment:
Have you thought of providing something like the following interface instead?
```
shopify_apt_get_on_change '/etc/nginx/version' do
  version node['nginx']['version']
  datadog_event: {
    name: "",
    text: "",
    key: ""
  }
end
```
I think this is quite common Chef pattern.